newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -0,0 +1,1360 @@
+      newVariable,
+      serializeToJSObject(inheritedVariable)
+    );
+    props.variablesContainer.insert(
+      inheritedVariableName,
+      newVariable,
+      props.variablesContainer.count()
+    );
+    _saveToHistory();
+    setSelectedNodes([inheritedVariableName]);
+    setExpandedNodes([...expandedNodes, inheritedVariableName]);
+    newVariable.delete();
+  };
+
+  const onAdd = () => {
+    const addAtTopLevel =
+      selectedNodes.length === 0 ||
+      selectedNodes.some(node => node.startsWith(inheritedPrefix));
+
+    if (addAtTopLevel) {
+      const newName = insertInVariablesContainer(
+        props.variablesContainer,
+        'Variable',
+        null,
+        props.variablesContainer.count()
+      );
+      _saveToHistory();
+      setSelectedNodes([newName]);
+      return;
+    }
+
+    const targetNode = selectedNodes[0];
+    const {
+      name: targetVariableName,
+      lineage: targetLineage,
+    } = getVariableContextFromNodeId(targetNode, props.variablesContainer);
+    if (!targetVariableName) return;
+    const oldestAncestry = getOldestAncestryVariable(targetLineage);
+    let position;
+    if (!oldestAncestry) {
+      position = props.variablesContainer.getPosition(targetVariableName) + 1;
+    } else {
+      position = props.variablesContainer.getPosition(oldestAncestry.name) + 1;
+    }
+    const newName = insertInVariablesContainer(
+      props.variablesContainer,
+      'Variable',
+      null,
+      position
+    );
+    _saveToHistory();
+    setSelectedNodes([newName]);
+  };
+
+  const renderVariableAndChildrenRows = ({
+    name,
+    variable,
+    parentNodeId,
+    parentVariable,
+    isInherited,
+  }: {|
+    name: string,
+    variable: gdVariable,
+    parentNodeId?: string,
+    parentVariable?: gdVariable,
+    isInherited: boolean,
+  |}) => {
+    const isCollection = isCollectionVariable(variable);
+    const type = variable.getType();
+
+    const depth = parentNodeId ? parentNodeId.split(separator).length : 0;
+    const isTopLevel = depth === 0;
+    const shouldWrap =
+      isNarrow ||
+      (!containerWidth
+        ? false
+        : containerWidth <= 750
+        ? depth >= 5
+        : containerWidth <= 850
+        ? depth >= 6
+        : containerWidth <= 950
+        ? depth >= 7
+        : depth >= 8);
+
+    let parentType = null;
+    let nodeId;
+    if (!parentNodeId) {
+      if (isInherited) {
+        nodeId = `${inheritedPrefix}${name}`;
+      } else {
+        nodeId = name;
+      }
+    } else {
+      nodeId = `${parentNodeId}${separator}${name}`;
+    }
+    if (!!parentVariable) {
+      parentType = parentVariable.getType();
+    }
+    const isSelected = selectedNodes.includes(nodeId);
+    const overwritesInheritedVariable =
+      isTopLevel &&
+      !isInherited &&
+      props.inheritedVariablesContainer &&
+      props.inheritedVariablesContainer.has(name);
+
+    if (
+      !!searchText &&
+      !name
+        .normalize('NFD')
```

Comment:
Is normalizing really useful?

Reply:
Seems like a standard of Web: When I search for `Café` in Maps, I also want `cafe` results

Reply:
I see, that would be helpful (and could be extracted into a `Utils/Search.js` file that contains a `normalize` function). Though when I do `"café".normalize('NFD')`, I still get `café` in a JS console?

Reply:
Oh yeah I forgot to remove diacritics

Reply:
1 point for Gryffindor (If in an util, I wouldn't have forgotten it)